app/actions/handlers.py

```diff
@@ -0,0 +1,138 @@
+import httpx
+import logging
+import stamina
+import pydantic
+
+import app.services.gundi as gundi_tools
+import app.actions.client as client
+
+from datetime import datetime, timezone, timedelta
+
+from app.actions.client import LotekException, LotekConnectionException
+from app.actions.configurations import get_auth_config, AuthenticateConfig, PullObservationsConfig
+from app.services.activity_logger import activity_logger
+from app.services.state import IntegrationStateManager
+
+logger = logging.getLogger(__name__)
+state_manager = IntegrationStateManager()
+
+async def action_auth(integration, action_config: AuthenticateConfig):
+    logger.info(f"Executing auth action with integration {integration} and action_config {action_config}...")
+    try:
+        token = await client.get_token(integration, action_config)
+    except LotekConnectionException:
+        logger.exception(f"Auth unsuccessful for integration {integration.id}. Lotek returned 400 (bad request)")
+        return {"valid_credentials": False, "message": "Invalid credentials"}
+    except httpx.HTTPError as e:
+        logger.exception(f"Auth action failed for integration {integration.id}. Exception: {e}")
+        return {"error": "An internal error occurred while trying to test credentials. Please try again later."}
+    else:
+        if token:
+            logger.info(f"Auth successful for integration '{integration.name}'. Token: '{token}'")
+            return {"valid_credentials": True}
+        else:
+            logger.error(f"Auth unsuccessful for integration {integration}.")
+            return {"valid_credentials": False}
+
+def transform(p: client.LotekPosition, integration):
+    try:
+        if p.Longitude and p.Latitude:
+            data = {
+                "source": p.DeviceID,
+                "source_name": p.DeviceID,
+                'type': 'tracking-device',
+                "recorded_at": ensure_timezone_aware(p.RecDateTime).isoformat(),
+                "location": {
+                    "lat": p.Latitude,
+                    "lon": p.Longitude
+                },
+                "additional": p.dict(exclude={'DeviceID', 'Latitude', 'Longitude', 'RecDateTime'})
+            }
+            return data
+        else:
+            logger.warning(f'Ignoring bad Lotek point: {p}')
+    except Exception as ex:
+        logger.error(f"Failed to parse Lotek point: {p}. Exception: {ex}", extra={
+            "attention_needed": True,
+            "integration_id": integration.id,
+            "integration_type": "lotek"
+        })
+        raise
```

Comment:
Remember that whatever you put in `extra` won't show up in gcp until we switch to use their own logger client. Maybe we can add the integration id in the message for now.